linkerd/app/inbound/src/policy/mod.rs

```diff
@@ -69,8 +84,19 @@ impl AllowPolicy {
     }
 
     #[inline]
-    pub(crate) fn server_labels(&self) -> Labels {
-        self.server.borrow().labels.clone()
+    pub fn dst_addr(&self) -> OrigDstAddr {
+        self.dst
+    }
+
+    #[inline]
+    pub fn server_label(&self) -> ServerLabel {
+        ServerLabel(self.server.borrow().name.clone())
+    }
+
+    async fn changed(&mut self) {
+        if self.server.changed().await.is_err() {
+            futures::future::pending::<()>().await;
+        }
```

Comment:
maybe worth a comment explaining why this should never complete if the watch was dropped? at a glance it seemed slightly scary, but I think I understand why we would want that --- we don't ever want to tear down those connections if the policy went away?